### ISSUE TITLE
WP-CLI returns cart info/shipping tax for first response after 'wp cache flush' is performed.

### ISSUE THREAD

**Issue by rcojoe** [woocommerce/woocommerce#18906]
## Prerequisites

- [x ] I have searched for similar issues in both open and closed tickets and cannot find a duplicate
- [x ] The issue still exists against the latest `master` branch of WooCommerce on Github (this is **not** the same version as on WordPress.org!)
- [x ] I have attempted to find the simplest possible steps to reproduce the issue
- [ ] I have included a failing test as a pull request (Optional)

## Steps to reproduce the issue

1. I issue the following commands:

```

localhost:~$ wp cache flush
Success: The cache was flushed.
localhost:~$ wp wc product_variation list 26765 --field=id --format=json
Cart Page
Cart Total 110.00
taxes Array
(
    [1] => 10.18
)

shipping class None
shipping rates 0
shipping tax
shipping rates after 0
shipping tax after
shipping rates 0
shipping tax
shipping rates after 0
shipping tax after
shipping rates 15
shipping tax
shipping rates after 15
shipping tax after
shipping rates 35
shipping tax
shipping rates after 35
shipping tax after
[26775,26774,26773,26771,26770,26769,26767,26768]
localhost:~$ wp wc product_variation list 26765 --field=id --format=json
[26775,26774,26773,26771,26770,26769,26767,26768]
localhost:~$
```

The first response after a cache clearing is the cart page info above along with the correct response.  The second request is the desired response.

Cart total dollar amount is always the same.  
My cache type is memcache

## Expected/actual behavior

[26775,26774,26773,26771,26770,26769,26767,26768]

## Isolating the problem

- [ ] This bug happens with only WooCommerce plugin active (not able to test)
- [x ] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/)
- [x ] I can reproduce this bug consistently using the steps above

## WordPress Environment

<details>
```
`
### WordPress Environment ###

Home URL: https://www.rockcreek.com
Site URL: https://www.rockcreek.com/gear
WC Version: 3.3.1
Log Directory Writable: ✔
WP Version: 4.9.3
WP Multisite: –
WP Memory Limit: 2 GB
WP Debug Mode: –
WP Cron: ✔
Language: en_US

### Server Environment ###

Server Info: Apache/2.4.18 (Ubuntu)
PHP Version: 7.0.22-0ubuntu0.16.04.1
PHP Post Max Size: 8 MB
PHP Time Limit: -1
PHP Max Input Vars: 1000
cURL Version: 7.47.0
OpenSSL/1.0.2g

SUHOSIN Installed: –
MySQL Version: 5.7.16
Max Upload Size: 8 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ❌ Your server does not have the SoapClient class enabled - some gateway plugins which use SOAP may not work as expected.
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 3.3.1
WC Database Prefix: R0cK_wp_
Total Database Size: 645.07MB
Database Data Size: 542.94MB
Database Index Size: 102.13MB
R0cK_wp_woocommerce_sessions: Data: 136.42MB + Index: 3.02MB
R0cK_wp_woocommerce_api_keys: Data: 0.02MB + Index: 0.03MB
R0cK_wp_woocommerce_attribute_taxonomies: Data: 0.02MB + Index: 0.02MB
R0cK_wp_woocommerce_downloadable_product_permissions: Data: 0.02MB + Index: 0.05MB
R0cK_wp_woocommerce_order_items: Data: 1.52MB + Index: 0.52MB
R0cK_wp_woocommerce_order_itemmeta: Data: 7.52MB + Index: 8.03MB
R0cK_wp_woocommerce_tax_rates: Data: 0.02MB + Index: 0.06MB
R0cK_wp_woocommerce_tax_rate_locations: Data: 0.02MB + Index: 0.05MB
R0cK_wp_woocommerce_shipping_zones: Data: 0.02MB + Index: 0.00MB
R0cK_wp_woocommerce_shipping_zone_locations: Data: 0.02MB + Index: 0.05MB
R0cK_wp_woocommerce_shipping_zone_methods: Data: 0.02MB + Index: 0.00MB
R0cK_wp_woocommerce_payment_tokens: Data: 0.02MB + Index: 0.02MB
R0cK_wp_woocommerce_payment_tokenmeta: Data: 0.02MB + Index: 0.03MB
R0cK_wp_woocommerce_log: Data: 0.02MB + Index: 0.02MB
order_detail: Data: 0.00MB + Index: 0.00MB
order_header: Data: 0.00MB + Index: 0.00MB
R0cK_wp_ak_twitter: Data: 0.14MB + Index: 0.05MB
R0cK_wp_blc_filters: Data: 0.00MB + Index: 0.00MB
R0cK_wp_blc_instances: Data: 0.23MB + Index: 0.05MB
R0cK_wp_blc_links: Data: 1.21MB + Index: 0.26MB
R0cK_wp_blc_synch: Data: 0.00MB + Index: 0.00MB
R0cK_wp_commentmeta: Data: 1.41MB + Index: 0.08MB
R0cK_wp_comments: Data: 7.99MB + Index: 4.25MB
R0cK_wp_cpf_custom_products: Data: 0.02MB + Index: 0.02MB
R0cK_wp_cp_feeds: Data: 0.02MB + Index: 0.00MB
R0cK_wp_facetwp_cache: Data: 26.02MB + Index: 0.00MB
R0cK_wp_facetwp_index: Data: 4.52MB + Index: 2.47MB
R0cK_wp_failed_jobs: Data: 0.17MB + Index: 0.00MB
R0cK_wp_followup_coupons: Data: 0.02MB + Index: 0.03MB
R0cK_wp_followup_coupon_logs: Data: 0.02MB + Index: 0.03MB
R0cK_wp_followup_customers: Data: 0.38MB + Index: 0.69MB
R0cK_wp_followup_customer_carts: Data: 1.52MB + Index: 0.14MB
R0cK_wp_followup_customer_notes: Data: 0.02MB + Index: 0.03MB
R0cK_wp_followup_customer_orders: Data: 0.31MB + Index: 0.20MB
R0cK_wp_followup_email_coupons: Data: 0.02MB + Index: 0.02MB
R0cK_wp_followup_email_excludes: Data: 0.02MB + Index: 0.05MB
R0cK_wp_followup_email_logs: Data: 0.02MB + Index: 0.05MB
R0cK_wp_followup_email_orders: Data: 0.13MB + Index: 0.09MB
R0cK_wp_followup_email_order_coupons: Data: 0.02MB + Index: 0.02MB
R0cK_wp_followup_email_tracking: Data: 0.02MB + Index: 0.08MB
R0cK_wp_followup_followup_history: Data: 0.05MB + Index: 0.05MB
R0cK_wp_followup_order_categories: Data: 0.23MB + Index: 0.31MB
R0cK_wp_followup_order_items: Data: 0.36MB + Index: 0.59MB
R0cK_wp_followup_subscribers: Data: 0.02MB + Index: 0.03MB
R0cK_wp_followup_subscribers_to_lists: Data: 0.02MB + Index: 0.03MB
R0cK_wp_followup_subscriber_lists: Data: 0.02MB + Index: 0.03MB
R0cK_wp_ks_giveaways_contestant: Data: 0.08MB + Index: 0.06MB
R0cK_wp_ks_giveaways_entry: Data: 0.08MB + Index: 0.11MB
R0cK_wp_ks_giveaways_winner: Data: 0.02MB + Index: 0.02MB
R0cK_wp_links: Data: 0.00MB + Index: 0.00MB
R0cK_wp_mailchimp_carts: Data: 0.02MB + Index: 0.00MB
R0cK_wp_omega_index_status: Data: 103.67MB + Index: 0.00MB
R0cK_wp_options: Data: 1.82MB + Index: 0.10MB
R0cK_wp_pmxe_exports: Data: 0.14MB + Index: 0.00MB
R0cK_wp_pmxe_google_cats: Data: 0.39MB + Index: 0.00MB
R0cK_wp_pmxe_posts: Data: 2.52MB + Index: 0.00MB
R0cK_wp_pmxe_templates: Data: 0.02MB + Index: 0.00MB
R0cK_wp_pmxi_files: Data: 0.02MB + Index: 0.00MB
R0cK_wp_pmxi_history: Data: 0.08MB + Index: 0.00MB
R0cK_wp_pmxi_imports: Data: 1.52MB + Index: 0.00MB
R0cK_wp_pmxi_posts: Data: 2.52MB + Index: 0.00MB
R0cK_wp_pmxi_templates: Data: 0.02MB + Index: 0.00MB
R0cK_wp_postmeta: Data: 110.52MB + Index: 65.26MB
R0cK_wp_posts: Data: 30.82MB + Index: 8.04MB
R0cK_wp_queue: Data: 8.52MB + Index: 0.00MB
R0cK_wp_relevanssi: Data: 15.52MB + Index: 0.00MB
R0cK_wp_relevanssi_log: Data: 35.56MB + Index: 0.00MB
R0cK_wp_relevanssi_stopwords: Data: 0.02MB + Index: 0.00MB
R0cK_wp_snippets: Data: 0.02MB + Index: 0.00MB
R0cK_wp_termmeta: Data: 0.69MB + Index: 0.44MB
R0cK_wp_terms: Data: 0.14MB + Index: 0.25MB
R0cK_wp_term_relationships: Data: 0.83MB + Index: 1.92MB
R0cK_wp_term_taxonomy: Data: 0.13MB + Index: 0.17MB
R0cK_wp_usermeta: Data: 1.55MB + Index: 1.10MB
R0cK_wp_useronline: Data: 0.00MB + Index: 0.00MB
R0cK_wp_users: Data: 0.07MB + Index: 0.09MB
R0cK_wp_wc_download_log: Data: 0.02MB + Index: 0.03MB
R0cK_wp_wc_webhooks: Data: 0.02MB + Index: 0.02MB
R0cK_wp_woocommerce_recommendations: Data: 3.52MB + Index: 0.00MB
R0cK_wp_woocommerce_session_activity: Data: 25.05MB + Index: 0.00MB
R0cK_wp_yoast_seo_links: Data: 0.02MB + Index: 0.02MB
R0cK_wp_yoast_seo_meta: Data: 0.05MB + Index: 0.00MB
rcprice: Data: 1.67MB + Index: 0.88MB
rcprice_changes: Data: 0.00MB + Index: 0.00MB
rcprice_old: Data: 1.67MB + Index: 1.19MB
rcqty: Data: 3.00MB + Index: 0.88MB

### Post Type Counts ###

acf: 1
amn_smtp: 1
attachment: 10259
customize_changeset: 6
custom_css: 2
easyimageslider: 1
follow_up_email: 2
grid_template: 7
ks_giveaway: 2
ml-slide: 6
ml-slider: 3
nav_menu_item: 16
page: 80
post: 1537
product: 3085
product_variation: 20764
revision: 1769
rootsrated_revision: 1
scheduled-action: 186
shop_coupon: 9
shop_order: 5980
shop_order_refund: 295
wc_order_email: 5
wc_order_status: 10

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (44) ###

Akismet Anti-Spam: by Automattic – 4.0
Categories in Hierarchical Order: by Kamal Agrawal
Amit Sonkhiya – 1.2

Cloudflare: by John Wineman
Furkan Yilmaz
Junade Ali (Cloudflare Team) – 3.3.2

Code Snippets: by Shea Bunge – 2.9.4
Display Categories Widget: by Suresh Baskaran – 2.1
Display Widgets: by Strategy11 – 2.05
Disqus Comment System: by Disqus  – 2.87
Google Tag Manager for Wordpress: by Thomas Geiger – 1.7.2
Facebook for WooCommerce: by Facebook – 1.6.6 – Not tested with the active version of WooCommerce
FacetWP - Cache: by FacetWP
LLC – 1.5.0

FacetWP: by FacetWP
LLC – 3.1.1

Get The Image: by Justin Tadlock – 1.1.0
Google XML Sitemaps: by Arne Brachhold – 4.0.9
Omega Instant Search For WooCommerce: by Omega Search – 1.6.10 – Not tested with the active version of WooCommerce
Page Links To: by Mark Jaquith – 2.9.9
Regenerate Thumbnails: by Alex Mills (Viper007Bond) – 3.0.2
RootsRated Compass: by rootsratedmedia – 1.3.2
Title Toggle for Storefront Theme: by Wooassist – 1.2.3
SumoMe: by SumoMe – 1.27
TaxJar - Sales Tax Automation for WooCommerce: by TaxJar – 1.5.4 – Not tested with the active version of WooCommerce
UpdraftPlus - Backup/Restore: by UpdraftPlus.Com
DavidAnderson – 2.13.6.1

Wc Bulk Product Delete: by Varun Sridharan – 0.3 – Not tested with the active version of WooCommerce
WC Duplicate Order: by Jamie Gill – 1.4.2 – Not tested with the active version of WooCommerce
WooSwatches - Woocommerce Color or Image Variation Swatches: by parbat chaudhari – 2.3.6 – Not tested with the active version of WooCommerce
WooCommerce Custom Payment Gateway Pro: by WPRuby – 1.3.8 – Not tested with the active version of WooCommerce
Follow-Up Emails: by WooCommerce – 4.6.4
WooCommerce Amazon Pay Gateway: by WooCommerce – 1.8.1
WooCommerce PayPal Powered by Braintree Gateway: by WooCommerce – 2.1.1
WooCommerce Google Trusted Stores Integration: by enollo – 1.0.0 – Not tested with the active version of WooCommerce
WooCommerce Order Status Manager: by SkyVerge – 1.9.0
WooCommerce Product CSV Import Suite: by WooCommerce – 1.10.17
WooCommerce Recommendation Engine: by Lucas Stark – 3.1.7 – 3.1.9 is available – Not tested with the active version of WooCommerce
WooCommerce Shipment Tracking: by WooCommerce – 1.6.10
WooCommerce: by Automattic – 3.3.1
Yoast SEO: by Team Yoast – 6.2
WP All Export: by Soflyy – 1.1.5
WP All Import Pro: by Soflyy – 4.5.0
WP Mail SMTP: by WPForms – 1.2.2
WP Super Cache: by Automattic – 1.5.9
WP All Import - WooCommerce Add-On Pro: by Soflyy – 2.3.8 – Not tested with the active version of WooCommerce
WPSSO Schema JSON-LD Markup Pro: by JS Morisset – 1.21.0
WPSSO Strip Schema Microdata: by JS Morisset – 1.2.7
WPSSO Update Manager: by JS Morisset – 1.8.2
WPSSO Core Pro: by JS Morisset – 3.52.1-dev.2

### Settings ###

API Enabled: ✔
Force SSL: ✔
Currency: USD ($)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-catalog (exclude-from-catalog-2)
exclude-from-search (exclude-from-search)
exclude-from-search (exclude-from-search-2)
featured (featured)
outofstock (outofstock)
outofstock (outofstock-2)
rated-1 (rated-1)
rated-1 (rated-1-2)
rated-2 (rated-2)
rated-2 (rated-2-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-4 (rated-4-2)
rated-5 (rated-5)
rated-5 (rated-5-2)


### WC Pages ###

Shop base: #26557 - /shop
Cart: #26558 - /cart
Checkout: #26559 - /checkout
My account: #26560 - /my-account
Terms and conditions: ❌ Page not set

### Theme ###

Name: Whiteboard Child Theme
Version: 2.1.0.1476111885
Author URL: https://woocommerce.com/
Child Theme: ✔
Parent Theme Name: Storefront
Parent Theme Version: 2.2.7
Parent Theme Author URL: https://woocommerce.com/
WooCommerce Support: ✔

### Templates ###

Overrides: whiteboard/woocommerce/archive-product.php version 2.0.0 is out of date. The core version is 3.3.0
whiteboard/woocommerce/content-product.php
whiteboard/woocommerce/emails/admin-new-order.php

Outdated Templates: ❌Learn how to update

### Braintree (Credit Card) ###

Environment: Production
Tokenization Enabled: –
Debug Mode: Off

### Braintree (PayPal) ###

Environment: Production
Tokenization Enabled: –
Debug Mode: Off
`
```
</details>

**Comment by mikejolley** [comment]
This is not happening on my install. It looks like debugging output (?) - maybe you've been adding something to your install or a plugin somewhere?

Search our source code for `shipping rates after` - you'll find no results.

Closing as non-reproducable.

**Comment by rcojoe** [comment]
@mikejolley  You were right.  This was caused by an error_log("Cart Page"); etc. entry  that I had set up in the functions.php.  
add_filter( 'woocommerce_package_rates', 'adjust_shipping_rate', 10, 2 );

User 31  just happens to have 110 dollars in their shopping cart but I still am not sure why this  error log info is showing up here.  I have lots of other error_log entries and they don't show up in my CLI replies.  

Joe
